### ISSUE TITLE
Is it possible to dagify a graph based on paths?

### ISSUE THREAD

**Issue by hgibling** [vgteam/vg#2345]
Hello,

I made a graph from 36 related sequences using `msga`. The resulting graph was cyclic, which was unexpected and not what I want (though the input sequences are somewhat repetitive). Is it possible to either specify within `msga` to make it acyclic, or to unroll the cycles based on the embedded paths?

I see a few `mod` options that are almost what I want. I've tried `--break-cycle`, but that didn't update the embedded paths. `--dagify-step, --dagify-to, and --dagify-len-max` seem too specific, as the amount of unrolling needed is probably different for each path.

Is there a way to do this?

**Comment by ekg** [comment]
Yes, you can force more linear alignments. Please see my thesis for the
correct command line. There is an example with vg msga of the MHC in
chapter 3, page 136. https://www.repository.cam.ac.uk/handle/1810/294516

I think you want to set -J to a large number. Maybe -J 10000 or
something like that. You can also increase the length of the chunks in the
chunked alignment with -w. I think -J is more important for this.

Another option is to use seqwish and set -r 1. This also prevents collapse,
but the graph may still be cyclic.

**Comment by ekg** [comment]
You'll have to try different parameters for this. I'm curious what works.
Please let me know.

**Comment by hgibling** [comment]
So I tested combinations of `-w {100, 1000, 10000, 100000, 1000000}` and `-J {100, 1000, 10000, 100000, 1000000}`. `-J` didn't seem to have any effect. Graphs became acyclic at `-w >= 10000`. 

My sequences range in length from 672-1512bp long, with 84bp similarly repetitive blocks.
I also tested the parameters with 1k and 10k flanking sequences on my sequences of interest. Making a graph for the 1k flank sequences didn't work using any combination of parameters ("Floating point exception"). For the 10k flank sequences, `w 10000` didn't work, but `-w >= 100000` produced acyclic graphs. The flanking seequences helped get the graph into a reasonable layout (without flanks there were a lot of nodes with 'N's, including several with only a single 'N'). 

Overall I think I have something I can work with now. Thanks!
